### PR TITLE
Grammar

### LINKED ISSUE #645: Typo in attack tutorial
https://github.com/zaproxy/zap-hud/blob/29613bfbab9e18dff9377b7416cd5c41eaad4782/src/main/zapHomeFiles/hudtutorial/en_GB/AttackMode.html#L13 "is is"
Thanks Manie :)

### PR DESCRIPTION
**Pull request by jsoref**
Most changes are based on app.grammarly.com

Alt tags are my personal preference. I haven't taken the effort to make them match the items within the application. I'll do that after I get some progress on commits in both repositories.

Each commit is generally for a single change. This makes arguing about individual changes simple. If someone objects to a change, I'll drop it.

fixes #645

### CODE REVIEW

File: src/main/zapHomeFiles/hudtutorial/en_GB/AttackMode.html [hunk 1]
```diff
@@ -7,12 +7,12 @@
 <body>
 <div class="roundContainer">
 	<h1>Attack Mode</H1>
-	The 'Attack Mode' <img src="crosshairs-grey.png"> tool performs the same attacks as an Active Scan but works in a subtly different way.
+	The 'Attack Mode' <img src="crosshairs-grey.png" alt="[crosshairs]"> tool performs the same attacks as an Active Scan but works in a subtly different way.
 	<p>
 	Instead of attacking the URLs known at the time it is started, it attacks any URLs that are in scope as soon as they
-	are discovered. It does not show any progress because it stays active with the <img src="crosshairs.png"> icon until is is stopped.
+	are discovered. It does not show any progress because it stays active with the <img src="crosshairs.png"> icon until it is stopped.
```
Comment (jsoref) [inline src/main/zapHomeFiles/hudtutorial/en_GB/AttackMode.html]: fixes #645

Comment (thc202) [reply src/main/zapHomeFiles/hudtutorial/en_GB/AttackMode.html]: Could you add to the PR description (to automatically close it)?

**Comment by kingthorin** [comment]
I’ll put together a list again and go through this later tonight.

**Comment by kingthorin** [comment]
I like the idea of alt tags but assuming they’re mostly used for accessibility concerns they probably shouldn’t have extra punctuation (the square brackets) and probably should describe the control or whatever.... note I’ve only skimmed a bit (this comment might not be universally applicable).

**Comment by jsoref** [comment]
I think this is a vaguely reasonable description of alt tag use: http://www.velvetblues.com/web-development-blog/proper-use-of-alt-text-in-html/

Really, getting alt tags right requires a concerted effort. And in principle it should be consistent between a demo and the actual UI.

This is somewhat to start a conversation. I included the changes here mostly because the tool I was using, app.grammarly.com, complained about not being able to understand sentences when the images weren't present, so essentially, I was talking to it.

In my regular daily life, I have a blind friend who uses a number of screen readers, and I used to be involved in both w3c and Mozilla as well as mobile devices including working with the screen reader groups for each of them (and I'll randomly try to use software using a screen reader).

**Comment by kingthorin** [comment]
Thanks, I'll keep that in mind while reviewing.

**Comment by kingthorin** [comment]
- [x] grammar: comma after introductory phrase
- [x] a11y: add alt tag
- [x] grammar: it's
- [x] grammar: comma-clause
- [x] spelling: javascript
- [x] grammar: simplify wording
- [x] grammar: simplify
- [x] a11y: alt icon
- [x] spelling: it
- [x] spelling: subsection
- [x] a11y: alt text
- [x] a11y: alt text
- [x] grammar: redundant preposition
- [x] grammar: shown on
- [x] spelling: formatting
- [x] grammar: pronoun problem
- [ ] grammar: simplifying sentence
  - This change doesn't work. I'm not sure what the problem was with the original.
- [x] spelling: read-only
- [x] grammar: comma after introductory phrase
- [x] a11y: alt tags
- [x] a11y: alt icons
- [x] grammar: reword
- [x] grammar: simplify
- [x] spelling: left-hand and right-hand
- [x] spelling: left-hand
- [x] spelling: right-hand
- [x] grammar: comma after introductory phrase
- [x] a11y: alt tag
- [x] a11y: alt tags
- [x] grammar: comma after introductory phrase
- [x] a11y: alt tag
- [x] spelling: left-hand
- [x] grammar: count agreement
- [x] spelling: -level
- [x] a11y: alt tag
- [x] spelling: -level
- [x] spelling: left-hand
- [x] grammar: commas for phrase
- [x] a11y: alt tags
- [x] spelling: you
- [x] grammar: comma after interrupter
- [x] a11y: alt tag
- [x] spelling: right-hand
- [x] spelling: -level
- [x] a11y: alt tags
- [x] a11y: alt tags
- [x] spelling: low-level
- [x] spelling: right-hand
- [x] a11y: alt tag
- [x] spelling: right-hand
- [x] spelling: it has not
- [x] grammar: article
- [x] grammar: rephrase
- [x] a11y: alt tag
- [x] a11y: better alt tags
- [x] a11y: alt tag
- [x] spelling: -clicking
- [x] spelling: -click
- [x] grammar: simplify
- [x] spelling: control over
- [x] grammar: reword
- [x] a11y: alt tag
- [x] spelling: right-click
- [ ] grammar: simplify
  - I see why you changed this, but I think given the text in the preceding paragraph it flows more clearly with "do own"...
- [x] spelling: let's
- [x] spelling: right-hand

File: src/main/zapHomeFiles/hudtutorial/en_GB/Comments.html [hunk 2]
```diff
@@ -24,7 +24,7 @@ <h1>Comments</H1>
 	Hovering over these icons will show the related comment and clicking on them will copy the comment to the clipboard.
 	<p>
 	Showing comments may well mess up the formatting of the target page, and in some cases may
-	break functionality if there are any scripts that rely on the DOM structure not changing. 
+	break functionality if there any scripts rely on the DOM structure not changing.
```
Comment (jsoref) [inline src/main/zapHomeFiles/hudtutorial/en_GB/Comments.html]: oops, the `there` before `any` was supposed to be dropped. *sigh*

File: src/main/zapHomeFiles/hudtutorial/en_GB/Warning.html [hunk 3]
```diff
@@ -12,7 +12,7 @@ <h1 class="warning">Obligatory Warning</H1>
 	You should only use ZAP (and therefore also the HUD) on applications that you either
 	own or have permission to attack.
 	<p>
-	The good news is that if you installed ZAP on your own machine then you do own this tutorial,
+	The good news is that if you installed ZAP on your own machine then you own this tutorial,
```
Comment (jsoref) [inline src/main/zapHomeFiles/hudtutorial/en_GB/Warning.html]: happy to drop this.

It was a grammarly suggestion (it really doesn't like certain extra words/longer phrases).

File: src/main/zapHomeFiles/hudtutorial/en_GB/Comments.html (hunk 2 above)
Comment (kingthorin) [reply src/main/zapHomeFiles/hudtutorial/en_GB/Comments.html]: Yup, it would make sense that way 👍

**Comment by jsoref** [comment]
you can use `master..FOO` or `develop..FOO` or similar.

**Comment by jsoref** [comment]
I suppose I could generate these checklists for you:
`hg log -r grammar%develop -T '- [ ] {desc}\n'`

File: src/main/zapHomeFiles/hudtutorial/en_GB/PageAlerts.html [hunk 4]
```diff
@@ -8,15 +8,15 @@
...
 	your current page without being distracted by issues on other pages. 
 	<p>
-	Alerts have an associated risk, and all alerts with the same level of risk are shown together:
+	Each alert has an associated risk, and all alerts with the same level of risk are shown together:
 	<ul>
-	<li><img src="page-alerts-high.png"> High Level Page Alerts
-	<li><img src="page-alerts-medium.png"> Medium Level Page Alerts
-	<li><img src="page-alerts-low.png"> Low Level Page Alerts
-	<li><img src="page-alerts-informational.png"> Informational Level Page Alerts
+	<li><img src="page-alerts-high.png" alt="[red flag]"> High-Level Page Alerts
```
Comment (thc202) [inline src/main/zapHomeFiles/hudtutorial/en_GB/PageAlerts.html]: These no longer match the labels shown in the dialogue, should those be changed?

Comment (jsoref) [reply src/main/zapHomeFiles/hudtutorial/en_GB/PageAlerts.html]: I'm not sure these ever matched:
https://github.com/zaproxy/zaproxy/blob/d91db825c988f2be808738983360fd31731815fd/zap/src/main/resources/org/zaproxy/zap/resources/Messages.properties#L1656-L1661

I can't find anything else that corresponds to this...

Comment (thc202) [reply src/main/zapHomeFiles/hudtutorial/en_GB/PageAlerts.html]: I meant in the HUD's UI:
https://github.com/zaproxy/zap-hud/blob/37c57279999707bdf3a644297d7a4e2ba25c8a70/src/other/resources/UIMessages/UIMessages.properties#L27

they are not the same either... thought they were, sorry.

**Comment by thc202** [comment]
Thank you!

File: src/main/zapHomeFiles/hudtutorial/en_GB/PageAlerts.html (hunk 4 above)
Comment (jsoref) [reply src/main/zapHomeFiles/hudtutorial/en_GB/PageAlerts.html]: To be fair, the fact that these three places do not agree is probably a bug.

And you were right in guessing that I wasn't checking.

**Pull request merged**